How many late days are you using for this assignment?  0

In [4]:
# Imports and pip installations (if needed)
%pip install pd
import pandas as pd
from sklearn.preprocessing import MinMaxScaler

Note: you may need to restart the kernel to use updated packages.


# Part 1: Load the dataset

In [5]:
df_categorical = pd.read_csv('chronic_kidney_disease_categorical.csv')
df_numerical = pd.read_csv('chronic_kidney_disease_numerical.csv')

In [6]:
df_numerical

,unique_id,age,bp,bgr,bu,sc,sod,pot,hemo,pcv,wbcc,rbcc
0,203694,38.0,80.0,99.0,19.0,0.5,147.0,3.5,13.6,44.0,7300.0,6.4
1,938027,43.0,60.0,108.0,25.0,1.0,144.0,5.0,17.8,43.0,7200.0,5.5
2,421471,37.0,60.0,111.0,35.0,0.8,135.0,4.1,16.2,50.0,5500.0,5.7
3,764115,70.0,90.0,144.0,125.0,4.0,136.0,4.6,12.0,37.0,8200.0,4.5
4,240975,47.0,80.0,95.0,35.0,0.9,140.0,4.1,NaN,NaN,NaN,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...
415,522912,56.0,80.0,165.0,55.0,1.8,NaN,NaN,13.5,40.0,11800.0,5.0
416,475200,64.0,100.0,163.0,54.0,7.2,140.0,4.6,7.9,26.0,7500.0,3.4
417,168242,48.0,60.0,112.0,44.0,1.2,142.0,4.9,14.5,44.0,9400.0,6.4
418,474407,21.0,90.0,107.0,40.0,1.7,125.0,3.5,8.3,23.0,12400.0,3.9


# Part 2: Analyze the Dataset

Refer to this: https://archive.ics.uci.edu/dataset/336/chronic+kidney+disease

Explain what the each data is in your own words. What are the features and labels? Are the features in the given datasets : categorical, numerical or both? Give 3 examples of categorical and numerical columns each (if they exist)

Answer: Features are the variables/columns in the data set. These are some examples of the features: bp:	blood pressure, sg: specific gravity, al:albumin, su:sugar, rbc:red blood cells

Labels are classifiers. It is to determine if the patient has chronic kidney diseases or not. The two labels ckd or nockd

# Part 3: Data Preprocessing

A fundamental skill in Machine Learning is mastering the art of data cleaning and preprocessing. In this assignment, you will learn and apply essential data cleaning techniques to transform a raw dataset into a clean, ready-to-use form which you can use for regression or classification tasks. By the end of this assignment, you'll have a fully clean dataset and a solid foundation in preparing data for various machine learning models.

## Part 3.1 : Drop Duplicate rows

Let's start by checking if the given datasets have any duplicate rows (same Unique Id). Use pandas to identify and remove these duplicate rows from the given dataset

In [7]:
# For the numerical dataset, check if there are duplicate rows in the dataset. If yes, print total number of duplicate rows
df_duplicated_numerical = df_numerical.duplicated().sum()
print(df_duplicated_numerical)

# Drop these duplicate rows
df_numerical.drop_duplicates(inplace=True)

# Repeat the same for categorical dataset. Print the duplicate rows and drop them
df_duplicated_categorical = df_categorical.duplicated().sum()
print(df_duplicated_categorical)

df_categorical.drop_duplicates(inplace=True)



30
30


## Part 3.2: Combine two differents datasets

A good skill to have is to know how to combine 2 different datasets.

Are all the unique ids are present in both datasets? Why do you think so? If not, what do the rows that are missing from one of the datasets look like in the combined table?

Answer:

In [8]:
# Merge the two given numerical and categorical datasets based on their unique_ID.

combined_df = pd.merge(df_numerical, df_categorical, on='unique_id') 

# Print the combined dataset
print(combined_df)

     unique_id   age     bp    bgr     bu   sc    sod  pot  hemo   pcv  ...  \
0       203694  38.0   80.0   99.0   19.0  0.5  147.0  3.5  13.6  44.0  ...   
1       938027  43.0   60.0  108.0   25.0  1.0  144.0  5.0  17.8  43.0  ...   
2       421471  37.0   60.0  111.0   35.0  0.8  135.0  4.1  16.2  50.0  ...   
3       764115  70.0   90.0  144.0  125.0  4.0  136.0  4.6  12.0  37.0  ...   
4       240975  47.0   80.0   95.0   35.0  0.9  140.0  4.1   NaN   NaN  ...   
..         ...   ...    ...    ...    ...  ...    ...  ...   ...   ...  ...   
385     522912  56.0   80.0  165.0   55.0  1.8    NaN  NaN  13.5  40.0  ...   
386     475200  64.0  100.0  163.0   54.0  7.2  140.0  4.6   7.9  26.0  ...   
387     168242  48.0   60.0  112.0   44.0  1.2  142.0  4.9  14.5  44.0  ...   
388     474407  21.0   90.0  107.0   40.0  1.7  125.0  3.5   8.3  23.0  ...   
389     926033  50.0   70.0  230.0   50.0  2.2    NaN  NaN  12.0  41.0  ...   

           pc         pcc          ba  htn   dm cad

## Part 3.3: Rows with Missing values

Removing missing values from a dataset is important for classification because it ensures the model is trained on complete and accurate data, leading to better performance and reliable predictions. Incomplete data can introduce bias and errors, negatively impacting the model's effectiveness.

In [9]:
# Calculate the percentage of rows that contain atleast one missing value

missing_percentage = combined_df.isnull().any(axis=1).mean() * 100

# Print %
print(f"Percentage of rows with at least one missing value: {missing_percentage:.2f}%")

# Drop these rows from the dataset
cleaned_combined_df = combined_df.dropna()

# Print the Dataset
print(cleaned_combined_df)



Percentage of rows with at least one missing value: 60.77%
     unique_id   age     bp    bgr    bu   sc    sod  pot  hemo   pcv  ...  \
0       203694  38.0   80.0   99.0  19.0  0.5  147.0  3.5  13.6  44.0  ...   
1       938027  43.0   60.0  108.0  25.0  1.0  144.0  5.0  17.8  43.0  ...   
2       421471  37.0   60.0  111.0  35.0  0.8  135.0  4.1  16.2  50.0  ...   
7       109053  64.0   60.0  106.0  27.0  0.7  150.0  3.3  14.4  42.0  ...   
13      349892  59.0   70.0  424.0  55.0  1.7  138.0  4.5  12.6  37.0  ...   
..         ...   ...    ...    ...   ...  ...    ...  ...   ...   ...  ...   
379     486397  20.0   70.0  123.0  44.0  1.0  135.0  3.8  14.6  44.0  ...   
381     462892  36.0   80.0   85.0  16.0  1.1  142.0  4.1  15.6  44.0  ...   
386     475200  64.0  100.0  163.0  54.0  7.2  140.0  4.6   7.9  26.0  ...   
387     168242  48.0   60.0  112.0  44.0  1.2  142.0  4.9  14.5  44.0  ...   
388     474407  21.0   90.0  107.0  40.0  1.7  125.0  3.5   8.3  23.0  ...   

    

## Part 3.4: Sort the dataset according to the Labels

In [10]:
# Sort the dataset according to the values in 'Target' column. Make sure reset the indices after sorting
sorted_combined_df = cleaned_combined_df.sort_values(by='Target').reset_index(drop=True)

# Print the dataset
print(sorted_combined_df)

     unique_id   age     bp    bgr     bu   sc    sod  pot  hemo   pcv  ...  \
0       474407  21.0   90.0  107.0   40.0  1.7  125.0  3.5   8.3  23.0  ...   
1       137148  73.0  100.0  295.0   90.0  5.6  140.0  2.9   9.2  30.0  ...   
2       484175  53.0   90.0   70.0  107.0  7.2  114.0  3.7   9.5  29.0  ...   
3       343710  61.0   80.0  173.0  148.0  3.9  135.0  5.2   7.7  24.0  ...   
4       532520  60.0   90.0  105.0   53.0  2.3  136.0  5.2  11.1  33.0  ...   
..         ...   ...    ...    ...    ...  ...    ...  ...   ...   ...  ...   
148     647552  61.0   70.0  120.0   29.0  0.7  137.0  3.5  17.4  52.0  ...   
149     125587  37.0   60.0  109.0   47.0  1.1  141.0  4.9  15.0  48.0  ...   
150     614376  57.0   80.0  119.0   17.0  1.2  135.0  4.7  15.4  42.0  ...   
151     853426  41.0   70.0  125.0   38.0  0.6  140.0  5.0  16.8  41.0  ...   
152     572140  55.0   80.0  104.0   28.0  0.9  142.0  4.8  17.3  52.0  ...   

           pc         pcc          ba  htn   dm  ca

## Part 3.5: Encoding Categorical data

In this step, we identify and process the categorical columns in the sorted dataset. We map each unique value in these columns to separate "dummy" columns.

For example, the column 'rbc' will be transformed into two columns 'rbc_normal' and 'rbc_abnormal'. If a row's value in 'rbc' is 'normal', the 'rbc_normal' column will be set to 1 and 'rbc_abnormal' will be set to 0.


**Note: Find a correct pandas function to do this **

In [11]:
# Write code here
categorical_columns = ['rbc', 'pc', 'pcc', 'ba', 'htn', 'dm', 'cad', 'appet', 'pe', 'ane', 'Target']

encoded_df = pd.get_dummies(cleaned_combined_df, columns=categorical_columns, drop_first=True)
encoded_df = encoded_df.astype(int)

# Print the dataset with encoded categorical data
print(encoded_df)

     unique_id  age   bp  bgr  bu  sc  sod  pot  hemo  pcv  ...  pc_normal  \
0       203694   38   80   99  19   0  147    3    13   44  ...          1   
1       938027   43   60  108  25   1  144    5    17   43  ...          1   
2       421471   37   60  111  35   0  135    4    16   50  ...          1   
7       109053   64   60  106  27   0  150    3    14   42  ...          1   
13      349892   59   70  424  55   1  138    4    12   37  ...          0   
..         ...  ...  ...  ...  ..  ..  ...  ...   ...  ...  ...        ...   
379     486397   20   70  123  44   1  135    3    14   44  ...          1   
381     462892   36   80   85  16   1  142    4    15   44  ...          1   
386     475200   64  100  163  54   7  140    4     7   26  ...          0   
387     168242   48   60  112  44   1  142    4    14   44  ...          1   
388     474407   21   90  107  40   1  125    3     8   23  ...          0   

     pcc_present  ba_present  htn_yes  dm_yes  cad_yes  appet_p

In the example we went through above, another solution is to have a single column for the binary variable. In the downstream modeling would this be equivalent? What effect would this have if the categorical variable could take more than 2 values? For example, let's say we have a categorical feature that is "type of condiment" that can take 5 separate values and we are trying to predict the rating of a particular sandwich.

Answer: If we have more categories we can further classify these results into more than binary 0 and 1s. In the healthcare example more classifications could classify a range of values for a variable and potentially analyze the progression of certain variables. For example 3 categories could determine low heatly and high categories for a variable like blood pressure. This is similar to categorizing with the condiments to ultimately determine the particular sandwich.

## Part 3.6 : Remove Outliers from Numerical Columns

Outliers can disproportionately influence the fit of a regression model, potentially leading to a model that does not generalize well therefore it is important that we remove outliers from the numerical columns of the dataset.

For this dataset, we define an outlier to be 3 times the standard deviation from the mean. Drop these outliers from the dataset

In [12]:
# Remove outliers
numerical_columns = ['age', 'bp', 'bgr', 'bu', 'sc', 'sod', 'pot', 'hemo', 'pcv', 'wbcc', 'rbcc']

# Print the dataset

for column in numerical_columns:
    mean = encoded_df[column].mean()
    std_dev = encoded_df[column].std()

    threshold_upper = mean + 3 * std_dev
    threshold_lower = mean - 3 * std_dev

    encoded_df = encoded_df[(encoded_df[column] <= threshold_upper) & (encoded_df[column] >= threshold_lower)]

print(encoded_df)

     unique_id  age   bp  bgr  bu  sc  sod  pot  hemo  pcv  ...  pc_normal  \
0       203694   38   80   99  19   0  147    3    13   44  ...          1   
1       938027   43   60  108  25   1  144    5    17   43  ...          1   
2       421471   37   60  111  35   0  135    4    16   50  ...          1   
7       109053   64   60  106  27   0  150    3    14   42  ...          1   
17      630177   58   80  131  18   1  141    3    15   53  ...          1   
..         ...  ...  ...  ...  ..  ..  ...  ...   ...  ...  ...        ...   
379     486397   20   70  123  44   1  135    3    14   44  ...          1   
381     462892   36   80   85  16   1  142    4    15   44  ...          1   
386     475200   64  100  163  54   7  140    4     7   26  ...          0   
387     168242   48   60  112  44   1  142    4    14   44  ...          1   
388     474407   21   90  107  40   1  125    3     8   23  ...          0   

     pcc_present  ba_present  htn_yes  dm_yes  cad_yes  appet_p

## Part 3.7 : Normalize the Numerical Columns

Normalizing numerical attributes ensures that all features contribute equally to the model by scaling them to a consistent range, which improves model performance and convergence. It prevents features with larger scales from disproportionately influencing the model's learning process.

In [13]:
# Normalize the all Numerical Attributes in the dataset.
scaler = MinMaxScaler()
encoded_df[numerical_columns] = scaler.fit_transform(encoded_df[numerical_columns])

# Print the dataset
print(encoded_df)




     unique_id       age    bp       bgr        bu     sc       sod       pot  \
0       203694  0.432432  0.50  0.124464  0.058824  0.000  0.900000  0.333333   
1       938027  0.500000  0.00  0.163090  0.098039  0.125  0.800000  1.000000   
2       421471  0.418919  0.00  0.175966  0.163399  0.000  0.500000  0.666667   
7       109053  0.783784  0.00  0.154506  0.111111  0.000  1.000000  0.333333   
17      630177  0.702703  0.50  0.261803  0.052288  0.125  0.700000  0.333333   
..         ...       ...   ...       ...       ...    ...       ...       ...   
379     486397  0.189189  0.25  0.227468  0.222222  0.125  0.500000  0.333333   
381     462892  0.405405  0.50  0.064378  0.039216  0.125  0.733333  0.666667   
386     475200  0.783784  1.00  0.399142  0.287582  0.875  0.666667  0.666667   
387     168242  0.567568  0.00  0.180258  0.222222  0.125  0.733333  0.666667   
388     474407  0.202703  0.75  0.158798  0.196078  0.125  0.166667  0.333333   

     hemo       pcv  ...  p

## Part 3.8: Remove Unnecessary columns

Are there any columns in this dataset which are not appropriate for modeling and predictions? Which column(s)? Justify their exclusion and remove them

Answer: In this case, the unique_id column is a prime candidate for exclusion because it serves as a unique identifier for each row but does not contribute to the prediction of the target variable.

In [14]:
#Remove that column
encoded_df = encoded_df.drop(columns=['unique_id'])

# Print the dataset
print(encoded_df.head(15))

         age   bp       bgr        bu     sc       sod       pot  hemo  \
0   0.432432  0.5  0.124464  0.058824  0.000  0.900000  0.333333   0.6   
1   0.500000  0.0  0.163090  0.098039  0.125  0.800000  1.000000   1.0   
2   0.418919  0.0  0.175966  0.163399  0.000  0.500000  0.666667   0.9   
7   0.783784  0.0  0.154506  0.111111  0.000  1.000000  0.333333   0.7   
17  0.702703  0.5  0.261803  0.052288  0.125  0.700000  0.333333   0.8   
18  0.229730  0.0  0.107296  0.091503  0.000  0.833333  1.000000   0.8   
19  0.581081  0.5  0.223176  0.143791  0.125  0.633333  0.333333   1.0   
20  0.554054  0.0  0.167382  0.098039  0.125  0.700000  0.666667   0.8   
22  0.567568  0.5  0.021459  0.078431  0.000  0.566667  1.000000   0.9   
23  0.364865  0.5  0.081545  0.058824  0.125  0.800000  1.000000   0.8   
24  0.364865  0.0  0.042918  0.098039  0.000  0.866667  0.333333   0.7   
35  0.689189  0.0  0.266094  0.052288  0.125  1.000000  0.666667   0.8   
37  0.486486  0.5  0.266094  0.091503 

## Part 3.9: Export the Cleaned Data

Now that you've completed these cleaning steps you should have a pandas dataframe which is much cleaner and ready for modeling. Our final step is to save our work. Export the DataFrame to a two new formats: csv and json.

In [15]:
# Export the dataframe to a new csv file
encoded_df.to_csv('cleaned_chronic_kidney_disease_data.csv', index=False)

# Export the dataframe to a new json file
encoded_df.to_json('cleaned_chronic_kidney_disease_data.json', orient='records', lines=True)

# Part 4: Data conversions with Large Language Models

One powerful use case of ChatGPT (and other generative language models) is cleaning and transforming data. In some cases, these models can directly manipulate loosely structured data that you provide to them into a standard format. In the other cases, you can often prompt the model to create a conversion or extraction script for you in python or Pandas and then run it on your own. 

In this part of the assignment you will prompt 383GPT to explore these capabilities.

## Part 4.1 GPT Data Manipulation

Take the cleaned dataset that you created in part three and output the top 15 rows of that dataset. Then copy the terminal output, open 383gpt and ask it to convert that output to a markdown table. Paste that markdown table in the cell bellow

Paste here: 

### Paste the markdown table here
|    | age      | bp     | bgr      | bu       | sc     | sod      | pot      | hemo     | pcv      | wbcc     | ... | pc_normal | pcc_present | ba_present | htn_yes |
|----|----------|--------|----------|----------|--------|----------|----------|----------|----------|----------|-----|-----------|-------------|-----------|---------|
| 0  | 0.432432 | 0.5    | 0.124464 | 0.058824 | 0.000  | 0.900000 | 0.333333 | 0.6      | 0.677419 | 0.241935 | ... | 1         | 0           | 0         | 0       |
| 1  | 0.500000 | 0.0    | 0.163090 | 0.098039 | 0.125  | 0.800000 | 1.000000 | 1.0      | 0.645161 | 0.233871 | ... | 1         | 0           | 0         | 0       |
| 2  | 0.418919 | 0.0    | 0.175966 | 0.163399 | 0.000  | 0.500000 | 0.666667 | 0.9      | 0.870968 | 0.096774 | ... | 1         | 0           | 0         | 0       |
| 3  | 0.783784 | 0.0    | 0.154506 | 0.111111 | 0.000  | 1.000000 | 0.333333 | 0.7      | 0.612903 | 0.306452 | ... | 1         | 0           | 0         | 0       |
| 4  | 0.702703 | 0.5    | 0.261803 | 0.052288 | 0.125  | 0.700000 | 0.333333 | 0.8      | 0.967742 | 0.201613 | ... | 1         | 0           | 0         | 0       |
| 5  | 0.229730 | 0.0    | 0.107296 | 0.091503 | 0.000  | 0.833333 | 1.000000 | 0.8      | 0.935484 | 0.161290 | ... | 1         | 0           | 0         | 0       |
| 6  | 0.581081 | 0.5    | 0.223176 | 0.143791 | 0.125  | 0.633333 | 0.333333 | 1.0      | 0.580645 | 0.104839 | ... | 1         | 0           | 0         | 0       |
| 7  | 0.554054 | 0.0    | 0.167382 | 0.098039 | 0.125  | 0.700000 | 0.666667 | 0.8      | 0.677419 | 0.083064 | ... | 1         | 0           | 0         | 0       |
| 8  | 0.567568 | 0.5    | 0.021459 | 0.078431 | 0.000  | 0.566667 | 1.000000 | 0.9      | 0.677419 | 0.243710 | ... | 1         | 0           | 0         | 0       |
| 9  | 0.364865 | 0.5    | 0.081545 | 0.058824 | 0.125  | 0.800000 | 1.000000 | 0.8      | 0.677419 | 0.206452 | ... | 1         | 0           | 0         | 0       |
| 10 | 0.364865 | 0.0    | 0.042918 | 0.098039 | 0.000  | 0.866667 | 0.333333 | 0.7      | 0.677419 | 0.308065 | ... | 1         | 0           | 0         | 0       |
| 11 | 0.689189 | 0.0    | 0.266094 | 0.052288 | 0.125  | 1.000000 | 0.666667 | 0.8      | 0.677419 | 0.406452 | ... | 1         | 0           | 0         | 0       |
| 12 | 0.486486 | 0.5    | 0.266094 | 0.091503 | 0.000  | 0.666667 | 0.666667 | 0.7      | 0.677419 | 0.281613 | ... | 1         | 0           | 0         | 0       |
| 13 | 0.905405 | 0.0    | 0.244635 | 0.248366 | 0.000  | 1.000000 | 0.333333 | 0.8      | 0.677419 | 0.430968 | ... | 1         | 0           | 0         | 0       |
| 14 | 0.986486 | 0.5    | 0.175966 | 0.222222 | 0.125  | 0.866667 | 0.333333 | 0.9      | 0.677419 | 0.087097 | ... | 1         | 0           | 0         | 0       |


** Caution: ** while language models can perform data conversions they also can * hallucinate * during this process, particularly for bigger datasets. Reflect on this below, how could you mitigate data conversion hallucinations from LLM conversions?

One way to mitigate data conversion hallucinations is to verify if the data that is converted by the LLM is the same as the cleaned data done through the analysis process. The disparities would show whether or not the LLM is reliable in converting the data.

## Part 4.2 GPT Pandas Prompting

In this section, you will prompt 383GPT to write pandas code manipulations for you.

After working with this data for awhile, we realized we're starting to forget the meanings of the abbreviated column names. Let's ask 383GPT to fix this for us. First, navigate to the [UCI dataset overview](https://archive.ics.uci.edu/dataset/336/chronic+kidney+disease) and copy the abbrevation to name mapping. Then, go to 383GPT and instruct the LLM to provide you with a pandas script to apply this renaming to all the columns of your dataset. Paste that code below and make any adjustments necessary to run it in your notebook.

In [16]:
# Code to rename all the columns in the dataset

encoded_df.rename(columns={
    'age_in_years': 'Age', 
    'bp_mmHg': 'Blood Pressure',
    'sg': 'Specific Gravity',
    'al': 'Albumin',
    'su': 'Sugar',
    'rbc': 'Red Blood Cells',
    'pc': 'Pus Cell',
    'pcc': 'Pus Cell clumps',
    'ba': 'Bacteria',
    'bgr_mg_dl': 'Blood Glucose Random',
    'bu_mg_dl': 'Blood Urea',
    'sc_mg_dl': 'Serum Creatinine',
    'sod_mEq_L': 'Sodium',
    'pot_mEq_L': 'Potassium',
    'hemo_gms': 'Hemoglobin',
    'pcv': 'Packed Cell Volume',
    'wc_cells_cumm': 'White Blood Cell Count',
    'rc_millions_cmm': 'Red Blood Cell Count',
    'htn': 'Hypertension',
    'dm': 'Diabetes Mellitus',
    'cad': 'Coronary Artery Disease',
    'appet': 'Appetite',
    'pe': 'Pedal Edema',
    'ane': 'Anemia',
    'class': 'Class'
}, inplace=True)

print(encoded_df)

          age    bp       bgr        bu     sc       sod       pot  hemo  \
0    0.432432  0.50  0.124464  0.058824  0.000  0.900000  0.333333   0.6   
1    0.500000  0.00  0.163090  0.098039  0.125  0.800000  1.000000   1.0   
2    0.418919  0.00  0.175966  0.163399  0.000  0.500000  0.666667   0.9   
7    0.783784  0.00  0.154506  0.111111  0.000  1.000000  0.333333   0.7   
17   0.702703  0.50  0.261803  0.052288  0.125  0.700000  0.333333   0.8   
..        ...   ...       ...       ...    ...       ...       ...   ...   
379  0.189189  0.25  0.227468  0.222222  0.125  0.500000  0.333333   0.7   
381  0.405405  0.50  0.064378  0.039216  0.125  0.733333  0.666667   0.8   
386  0.783784  1.00  0.399142  0.287582  0.875  0.666667  0.666667   0.0   
387  0.567568  0.00  0.180258  0.222222  0.125  0.733333  0.666667   0.7   
388  0.202703  0.75  0.158798  0.196078  0.125  0.166667  0.333333   0.1   

     Packed Cell Volume      wbcc  ...  pc_normal  pcc_present  ba_present  \
0        

## Part 4.3 Augmenting our skills with prompting

In addition, we can also use 383GPT to convert our data manipulation operations between different data manipulation languages and libraries. For example let's prompt 383GPT to convert the following SQL query to a pandas query.

**SQL Query**
```sql
SELECT Target, COUNT(*) AS count
FROM your_table_name
GROUP BY Target;
```

Prompt 383GPT to convert this to a pandas query. Run this query below, then describe what it does. (If you're not familiar with SQL that is okay you need to only comment on the final resulting output.)

In [17]:
# Converted SQL to Pandas code

# Assuming 'df' is your DataFrame containing the data
result = encoded_df.groupby('Target_notckd').size().reset_index(name='count')

# Display the resulting DataFrame
print(result)





   Target_notckd  count
0              0     22
1              1    111


*Describe what the above code does here*
The output shows that there are 22 occurrences of 0 and 111 occurrences of 1 in the Target_notckd column. This means that there are 22 occurences that the target has Chronic Kidney Diseases and 111 occurences that the target does not have chronic kidney diseases 
